core/bin/sqlite_worker.rs

```diff
@@ -100,7 +105,16 @@ impl WorkerState {
                 }
             }
 
-            self.cleanup_inactive_databases().await;
+            // Run cleanup in background if enough time has passed.
+            if last_cleanup.elapsed() >= CLEANUP_INTERVAL {
+                let registry = self.registry.clone();
+                tokio::task::spawn(async move {
```

Comment:
you want `tokio::task::spawn_blocking`, otherwise it still uses the same event loop. Spawn blocking uses a separate green thread pool